explainer.md

```diff
@@ -0,0 +1,259 @@
+
+### Painted Content
+A painted representation of an element animated as an atomic unit/layer during the transition. The representation effectively provides the element's content as a rendered image. This is defined using the existing [element()](https://drafts.csswg.org/css-images-4/#element-notation) function. The image animated during the transition is live for elements in the incoming document as defined in the spec. And a static version is cached by the browser for elements in the outgoing document.
+
+This proposal outlines the following modifications to the element() spec for ease of implementation:
+* The snapshotted element must have paint containment (contain:paint) to ensure the element is the containing block for all positioned descendents and generates a stacking context.
+* The snapshotted element must disallow fragmentation (break-inside:avoid).
+
+Generating this for the root element needs to be refined. The following is the list of special cases to consider there :
+* The natural size for the generated image is the visual viewport bounds.
+* When creating the image, the element is drawn on a canvas with the background color of the document.
+
+### Computed Properties
+The DOM state required to display the list of images/layers generated above to close visual fidelity with their rendering in the actual DOM. This is limited to the transform mapping a shared element to its quad in viewport space and the element's border-box size. Similar to painted representation above, this state is live for elements in the incoming document and a static cached version for elements in the outgoing document.
+
+This proposal disallows a shared element to be nested inside another shared element to avoid preserving the hierarchy of these DOM elements and associated properties (transform, clip, effects inherited by descendents). This helps in limiting the scope for the first iteration of this feature.
+
+## Rendering Transition DOM
+The DOM representation referenced above is rendered and animated by creating pseudo elements with the [content](https://drafts.csswg.org/css-content/#content-property) property set to a shared element's painted content, the [transform](https://drafts.csswg.org/css-transforms-1/#propdef-transform) set to the viewport space transform and width/height set to the element's border-box size. These pseudo elements are drawn on top of the document by painting them in the DOM's [top layer](https://fullscreen.spec.whatwg.org/#top-layer) retaining their paint order in the actual DOM.
+
+# Design
+This section describes the high level design for this feature with a tentative API:
+
+## Outgoing Document
+The transition sequence starts with the prepare phase when the browser receives a trigger for a transition. Consider the following snippet of code:
+
+```js
+<div shared-id="shared-elem"></div>
+
+// MPA version.
+addEventListener("navigate", (event) => {
```

Comment:
Does setData cause a transition to occur? This API example is hard to understand.

Reply:
Hopefully the new example makes this clearer.